internal/dbunique/db_unique.go

```diff
@@ -32,195 +28,127 @@ var defaultUniqueStates = []rivertype.JobState{ //nolint:gochecknoglobals
 
 var defaultUniqueStatesStrings = sliceutil.Map(defaultUniqueStates, func(s rivertype.JobState) string { return string(s) }) //nolint:gochecknoglobals
 
+var jobStateBitPositions = map[rivertype.JobState]uint{ //nolint:gochecknoglobals
+	rivertype.JobStateAvailable: 7,
+	rivertype.JobStateCancelled: 6,
+	rivertype.JobStateCompleted: 5,
+	rivertype.JobStateDiscarded: 4,
+	rivertype.JobStatePending:   3,
+	rivertype.JobStateRetryable: 2,
+	rivertype.JobStateRunning:   1,
+	rivertype.JobStateScheduled: 0,
+}
```

Comment:
These numbers correspond to the postgres bitshift numbers, which go from left to right. That's why the corresponding Go bit shifting code below shifts by `(7 - postgres_index)` when building up the bitmask.